components/support/guid/src/lib.rs

```diff
@@ -110,15 +110,54 @@ fn can_use_fast<T: ?Sized + AsRef<[u8]>>(bytes: &T) -> bool {
 }
 
 impl Guid {
+    /// Create a guid from a `str`.
+    #[inline]
+    pub fn new(s: &str) -> Self {
+        Guid::from_slice(s.as_ref())
+    }
+
+    /// Create an empty guid. Usable as a constant.
+    #[inline]
+    pub const fn empty() -> Self {
+        Guid(Repr::Fast(FastGuid {
+            len: 0,
+            data: [0u8; MAX_FAST_GUID_LEN],
+        }))
+    }
+
+    /// Create a random guid (of 12 base64url characters). Requires the `random`
+    /// feature.
+    #[cfg(feature = "random")]
+    pub fn random() -> Self {
+        let mut bytes = [0u8; 9];
+        // TODO: investigate if we should replace this with something infallible
+        // (from e.g. `rand`)
+        rc_crypto::rand::fill(&mut bytes).unwrap();
+
+        // Note: only first 12 bytes are used, but remaining are required to
+        // build the FastGuid
+        let mut output = [0u8; MAX_FAST_GUID_LEN];
+
+        let bytes_written =
+            base64::encode_config_slice(&bytes, base64::URL_SAFE_NO_PAD, &mut output[..12]);
+
+        debug_assert!(bytes_written == 12);
+
+        Guid(Repr::Fast(FastGuid {
+            len: 12,
+            data: output,
+        }))
+    }
+
     /// Convert `b` into a `Guid`.
     #[inline]
-    fn from_string(s: String) -> Self {
+    pub fn from_string(s: String) -> Self {
```

Comment:
I took out these `pub` in my last PR. Do we need these functions to be called from other crates?

Reply:
Yeah, it can be a pain to make `From::from` know what you're targeting sometimes (type ascription would help that, if rust ever gets it), and so IMO it's nice to expose non-generic versions of these.

Reply:
I see. Thanks!